src/main/java/org/mozilla/jss/ssl/javax/JSSEngine.java

```diff
@@ -935,6 +953,71 @@ public boolean getWantClientAuth() {
         return want_client_auth;
     }
 
+    /**
+     * Sets a SSLSocketListener on this object.
+     */
+    public void setListeners(Collection<? extends EventListener> new_listeners) {
+        listeners = new_listeners;
+    }
+
+    /**
+     * Gets the set of SSLSocketListeners on this object.
+     */
+    public Collection<? extends EventListener> getListeners() {
+        return listeners;
+    }
+
+    /**
+     * Fires any and all SSLSocketListeners on the specified alert received event.
+     *
+     * To be used by other implementations of JSSEngine.
+     */
+    protected void fireAlertReceived(SSLAlertEvent event) {
+	    System.err.println("JsSEngine.fireAlertReceived: " + event + " listeners: " + listeners);
```

Comment:
Probably should consider removing println's as this could become excessive.  Same with all the non-error debugs.